File: typescript/src/renderer.ts
import { Shape } from "./shapes/shape.js";

export class Renderer {
    private shape: Shape | null;

    constructor(shape?: Shape) {
        this.shape = shape ?? null; 
    }

    setShape(shape: Shape): void{
        this.shape = shape;
    } 

    draw(): void{
        if (this.shape === null) {
            console.log("no shape set");
            return;
        }
        const area: number = this.shape.computeArea();
        console.log("shape drawn \n", "Its area is " + area);
    }
}

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.Shape;

public class Renderer {
    private Shape shape;
    
    Renderer() {
        this.shape = null;
    }

    Renderer(Shape shape) {
        this.shape = shape;
    }

    public void setShape(Shape shape) {
        this.shape = shape;
    }

    void draw() {
        if (this.shape == null) {
            System.out.println("No shape set");
            return;
        }
        
        double area = shape.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}
